File: ClientApp/src/app/models/algorithm-measurement.ts
import { AlgorithmMeasurementResult } from "./algorithm-measurement-result";
import { AlgorithmMeasurementState } from "./algorithm-measurement-state";
import { BenchmarkAlgorithm } from "./benchmark-algorithm";

export interface AlgorithmMeasurement {
  algorithm: BenchmarkAlgorithm;
  parameter: number;
  state: AlgorithmMeasurementState;
  result: AlgorithmMeasurementResult | undefined;
}

File: Server/Models/AlgorithmMeasurement.cs
using System.ComponentModel.DataAnnotations;

namespace BenchmarkNow
{
    public class AlgorithmMeasurement
    {
        [Key]
        public long Id { get; set; }
        public Algorithm Algorithm { get; set; }
        public int Parameter { get; set; }
        public AlgorithmMeasurementResult Result { get; set; }
    }
}
